hypothesis-python/tests/ghostwriter/recorded/hypothesis_module_magic.txt

```diff
@@ -0,0 +1,100 @@
+# This test code was written by the `hypothesis.extra.ghostwriter` module
+# and is provided under the Creative Commons Zero public domain dedication.
+
+import hypothesis
+import typing
+from hypothesis import given, settings, strategies as st
+from random import Random
+
+
+@given(condition=st.from_type(object))
+def test_fuzz_assume(condition):
+    hypothesis.assume(condition=condition)
+
+
+@given(value=st.text())
+def test_fuzz_event(value):
+    hypothesis.event(value=value)
+
+
+@given(
+    specifier=st.from_type(hypothesis.strategies.SearchStrategy),
+    condition=st.functions(like=lambda *a: ..., returns=st.booleans()),
+    settings=st.one_of(st.none(), st.builds(settings)),
+    random=st.one_of(st.none(), st.builds(Random)),
+    database_key=st.one_of(st.none(), st.binary()),
+)
+def test_fuzz_find(specifier, condition, settings, random, database_key):
+    hypothesis.find(
+        specifier=specifier,
+        condition=condition,
+        settings=settings,
+        random=random,
+        database_key=database_key,
+    )
+
+
+@given(value=st.text())
+def test_fuzz_note(value):
+    hypothesis.note(value=value)
+
+
+@given(r=st.builds(Random))
+def test_fuzz_register_random(r):
+    hypothesis.register_random(r=r)
+
+
+@given(version=st.text(), blob=st.binary())
+def test_fuzz_reproduce_failure(version, blob):
+    hypothesis.reproduce_failure(version=version, blob=blob)
+
+
+@given(seed=st.from_type(typing.Hashable))
+def test_fuzz_seed(seed):
+    hypothesis.seed(seed=seed)
+
+
+@given(
+    parent=st.none(),
+    max_examples=st.just(not_set),
```

Comment:
I might miss something, but should the ghostwriter generate an import for `not_set`? E.g., `from hypothesis.utils.conventions import not_set`

Reply:
Adding `from hypothesis.utils.conventions import not_set` would indeed be better in this particular case.  However, I couldn't think of a heuristic that would get this to work without also adding unneeded imports for many other `just()` or `sampled_from()` strategies 😕 

I'm always open to suggestions, but had decided to leave it to a future PR for now - better to ship each part as it's ready than wait for perfection!

Reply:
Makes sense to me! :)